### DISCUSSION TITLE
AWS Glue - Null values in partition column.

### DISCUSSION THREAD

**Discussion by fyse-nassar** [aws/aws-sdk-cpp#2803] [Q&A] [answered]
It seems like glue isn't handling the getpartitions API correctly where a partition column has null value.
Example below, I am using the aws cli for simiplicity , which gives the same output as the SDK

My table data is structured as below in S3
```
s3://example-bucket/example_table/
├── int_partition_col=null/
│   ├── string_partition_col=null/
│   │   └── data-part-00001.csv
├── int_partition_col=1/
│   ├── string_partition_col=A/
│   │   └── data-part-00002.csv
└── int_partition_col=2/
    ├── string_partition_col=B/
    │   └── data-part-00003.csv

```
```
> aws glue get-partitions --database-name example_db --table-name example_table --expression "(int_partition_col >= 0)" ->
An error occurred (InvalidStateException) when calling the GetPartitions operation: For input string: "null" is not an integer.

> aws glue get-partitions --database-name example_db --table-name example_table --expression "(string_partition_col is null)" -> Returns empty

> aws glue get-partitions --database-name example_db --table-name example_table --expression "(string_partition_col = 'null')"-> works correctly

```  
So it seems like the null value is being considered as a string literal? But from the documentation [here](https://docs.aws.amazon.com/glue/latest/webapi/API_GetPartitions.html), it seems IS NULL etc are supported?

**Comment by jmklix** [comment, marked as answer]
Thanks for finding this and pointing it out. Since you also reproduced it in the CLI it looks like a problem with the model. I created an issue [here](https://github.com/aws/aws-sdk/issues/670) because it affects multiple sdks. I'm working to get the model updated and I will post any updates on the issue I linked above.
